lib/util/identifier.js

```diff
@@ -7,10 +7,26 @@ const looksLikeAbsolutePath = (maybeAbsolutePath) => {
 
 const normalizePathSeparator = (p) => p.replace(/\\/g, "/");
 
+/* A map from context dir strings to maps from identifier names to relative paths */
+const relativePaths = new Map();
```

Comment:
Could you put this cache in `compilation.cache.relativePaths`, so it doesn't leak memory?